cli/hse_cli.c

```diff
@@ -1447,6 +1449,97 @@ cli_hse_storage(struct cli_cmd *self, struct cli *cli)
     return sub_cmd->cmd_main(sub_cmd, cli);
 }
 
+static int
+cli_hse_strerror(struct cli_cmd *self, struct cli *cli)
+{
+    const struct cmd_spec spec = {
+        .usagev =
+            {
+                "[options] [--] <errorcode>",
+                NULL,
+            },
+        .optionv =
+            {
+                OPTION_HELP,
+                { "-v, --verbose", "Verbose output" },
+                { NULL },
+            },
+        .longoptv =
+            {
+                { "help", no_argument, 0, 'h' },
+                { "verbose", no_argument, 0, 'v' },
+                { NULL },
+            },
+        .configv =
+            {
+                { NULL },
+            },
+    };
+
+    const char *error_code_str;
+    bool        help = false;
+    bool        verbose = false;
+    char        c;
+    char        buf[1024];
+    int64_t     s_err;
+
+    if (cli_hook(cli, self, &spec))
+        return 0;
+
+    while (-1 != (c = cli_getopt(cli))) {
+        switch (c) {
+            case 'h':
+                help = true;
+                break;
+            case 'v':
+                verbose = true;
+                break;
+            default:
+                return EX_USAGE;
+        }
+    }
+
+    error_code_str = cli_next_arg(cli);
+
+    if (!error_code_str || help) {
+        int rc = help ? 0 : EX_USAGE;
+
+        cmd_print_help(self, help ? stdout : stderr);
+
+        if (!verbose) {
+            printf("\nUse -hv for more detail\n\n");
+            return rc;
+        }
+
+        printf("\nNote that if a file name was shown in the verbose output, it is reliable only\n");
+        printf("when the input error code was generated on this build.\n");
+
+        return rc;
+    }
+
+    if (parse_s64(error_code_str, &s_err)) {
+        fprintf(
+            stderr,
+            "%s: unable to parse"
+            " '%s' as a signed 64-bit"
+            " scalar value\n",
+            self->cmd_path,
+            error_code_str);
+        return EX_USAGE;
+    }
```

Comment:
This looks formatted for 24 column output.  Suggest not splitting the string constant and eliminating un-necessary newlines.